lib/xclarity_client/xclarity_base.rb

```diff
@@ -16,7 +16,11 @@ def initialize(conf, uri)
     def connection_builder(conf, uri)
       $lxca_log.info "XClarityClient::XClarityBase connection_builder", "Creating connection to #{conf.host + uri}" 
       #Building configuration
-      @conn = Faraday.new(url: conf.host + uri) do |faraday|
+      host = conf.host;
+      if !host.start_with?('https://') && !host.start_with?('http://')
+         host = 'https://' + host
```

Comment:
Even if they use HTTPS over another port I am not proposing to change the port information that was passed in. We are only adding https:// to the front of any URI that does not specify the protocol.  If they do specify a protocol & port, we will not modify the URI at all.